delphi-checks/src/main/java/au/com/integradev/delphi/checks/FormFmxCheck.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Sonar Delphi Plugin
+ * Copyright (C) 2024 Integrated Application Development
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package au.com.integradev.delphi.checks;
+
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+import org.sonar.check.Rule;
+import org.sonar.plugins.communitydelphi.api.ast.DelphiAst;
+import org.sonar.plugins.communitydelphi.api.ast.DelphiNode;
+import org.sonar.plugins.communitydelphi.api.ast.TypeDeclarationNode;
+import org.sonar.plugins.communitydelphi.api.check.DelphiCheck;
+import org.sonar.plugins.communitydelphi.api.check.DelphiCheckContext;
+import org.sonar.plugins.communitydelphi.api.directive.CompilerDirective;
+import org.sonar.plugins.communitydelphi.api.directive.ResourceDirective;
+import org.sonar.plugins.communitydelphi.api.token.DelphiToken;
+import org.sonar.plugins.communitydelphi.api.type.Type;
+
+@Rule(key = "FormFmx")
+public class FormFmxCheck extends DelphiCheck {
```

Comment:
There's a really significant overlap between these two checks - could it be refactored
as two children of a base check class? I worry SonarCloud will complain about the level
of duplication here.

Reply:
I feel like a base class for such simple checks might be overkill.
The core logic *could* be extracted out though, that is true.

I should set up [branch analysis](https://docs.sonarsource.com/sonarcloud/enriching/branch-analysis-setup/) so we can get some certainty about these metrics instead of speculating.